src/PrivateDns/PrivateDns/Models/PrivateDnsClient.cs

```diff
@@ -294,5 +294,292 @@ private static PSPrivateDnsVirtualNetworkLink ToPrivateDnsLink(VirtualNetworkLin
+                }
+
+                FillRecordsForType(properties, recordType, resourceRecords);
+            }
+            else
+            {
+                FillEmptyRecordsForType(properties, recordType);
+            }
+
+            return properties;
+        }
+
+        private static void FillRecordsForType(RecordSet properties, RecordType recordType, IReadOnlyList<PSPrivateDnsRecordBase> resourceRecords)
+        {
+            switch (recordType)
+            {
+                case RecordType.A:
+                    properties.ARecords = resourceRecords.Select(x => (Sdk.ARecord)(x as ARecord)?.ToMamlRecord()).ToList();
+                    break;
+                case RecordType.AAAA:
+                    properties.AaaaRecords = resourceRecords.Select(x => (Sdk.AaaaRecord)(x as AaaaRecord)?.ToMamlRecord()).ToList();
+                    break;
+                case RecordType.CNAME:
+                    if (resourceRecords.Count > 1)
+                    {
+                        throw new ArgumentException(ProjectResources.Error_AddRecordMultipleCnames);
+                    }
+
+                    properties.CnameRecord = (Sdk.CnameRecord)resourceRecords[0].ToMamlRecord();
+                    break;
+                case RecordType.MX:
+                    properties.MxRecords = resourceRecords.Select(x => (Sdk.MxRecord)(x as MxRecord)?.ToMamlRecord()).ToList();
+                    break;
+                case RecordType.PTR:
+                    properties.PtrRecords = resourceRecords.Select(x => (Sdk.PtrRecord)(x as PtrRecord)?.ToMamlRecord()).ToList();
+                    break;
+                case RecordType.SRV:
+                    properties.SrvRecords = resourceRecords.Select(x => (Sdk.SrvRecord)(x as SrvRecord)?.ToMamlRecord()).ToList();
+                    break;
+                case RecordType.TXT:
+                    properties.TxtRecords = resourceRecords.Select(x => (Sdk.TxtRecord)(x as TxtRecord)?.ToMamlRecord()).ToList();
+                    break;
+                case RecordType.SOA:
+                    properties.SoaRecord = (Sdk.SoaRecord)resourceRecords[0].ToMamlRecord();
+                    break;
+            }
+        }
+
+        private static void FillEmptyRecordsForType(RecordSet properties, RecordType recordType)
+        {
+            properties.AaaaRecords = recordType == RecordType.AAAA ? new List<Sdk.AaaaRecord>() : null;
+            properties.ARecords = recordType == RecordType.A ? new List<Sdk.ARecord>() : null;
+            properties.CnameRecord = recordType == RecordType.CNAME ? new Sdk.CnameRecord(String.Empty) : null;
+            properties.MxRecords = recordType == RecordType.MX ? new List<Sdk.MxRecord>() : null;
+            properties.PtrRecords = recordType == RecordType.PTR ? new List<Sdk.PtrRecord>() : null;
+            properties.SoaRecord = null;
```

Comment:
Again, why is SOA different from CNAME here?

Reply:
same as above, creation of soa records is not allowed in new cmdlet. so we don't need the check.